doc/typescript-target.md

```diff
@@ -30,88 +30,95 @@ The runtime is webpacked and sits in the dist folder. A .map file is also provid
+* visit the parse tree using a custom listener
+* visit the parse tree using a custom visitor
+* populate the grammar with production code (like AntLR3)
+  * _This approach is discouraged in Antlr4, Production code is target specific, which would prevent a grammar being used by multiple target languages_
  
 ## How do I create and run a visitor?
 
-You need to create a custom visitor and use it to visit the parse tree, as follows:
+You need to create an instance of custom visitor class and use it to visit the parse tree, as follows:
+
 ```typescript
+import { ParseTreeVisitor } from "antlr4";
+import { MyStartRuleContext} from "./MyLanguageParser";
+import MyLanguageVisitor from "./MyLanguageVisitor";
+import { parseMyLangauge } from "./typescript-parse";
 
-import { ParserRuleContext } from 'antlr4';
-import MyGrammarVisitor from './MyGrammarVisitor';
+type MyVisitResult = void;
 
-class CustomVisitor extends MyGrammarVisitor {
+class CustomVisitor
+    extends ParseTreeVisitor<MyVisitResult>
+    implements MyLanguageVisitor<MyVisitResult>
+{
 
-  visitChildren(ctx: ParserRuleContext) {
-    if (!ctx) {
-      return;
-    }
-    if (ctx.children) {
-      return ctx.children.map(child => {
-        if (child.children && child.children.length != 0) {
-          return child.accept(this);
-        } else {
-          return child.getText();
-        }
-      });
+    visitMyStartRule(ctx: MyStartRuleContext): MyVisitResult {
+        return this.visitChildren(ctx);
     }
-  }
+
 }
 
-tree.accept(new CustomVisitor());
+function parseAndVisit(codeToParse: string) {
+    const parseResult = parseMyLangauge(codeToParse);
+    const visitor = new CustomVisitor();
+    visitor.visit(parseResult);
+}
 ````
 
 ## How do I create and run a custom listener?
 
-You need to create a custom listener and use it to visit the parse tree, as follows:
+You need to create an instance of a custom listener class and use it to visit the parse tree, as follows:
 
 ```typescript
+import { ParseTreeWalker, ParseTreeListener } from "antlr4";
+import { MyStartRuleContext } from "./MyLanguageParser";
+import MyLanguageListener from "./MyLanguageListener";
+import { parseMyLangauge } from "./typescript-parse";
+
+class CustomListener
+    extends ParseTreeListener
+    implements MyLanguageListener
+{
+    enterMyStartRule(ctx: MyStartRuleContext) {
+        console.log("Enter Node 'myStartRule'");
+    }
+    exitMyStartRule(ctx: MyStartRuleContext) {
+        console.log("Exit Node 'myStartRule'");
+    }
 
-import { ParseTreeWalker } from 'antlr4';
-import MyGrammarListener from './MyGrammarListener';
-
-class MyTreeWalker extends MyGrammarListener {
-
-    exitMyStartRule = (ctx: MyStartRuleContext) => {
-        console.log("In MyStartRule");
-    };
-    
 }
 
-const walker = new MyTreeWalker();
-ParseTreeWalker.DEFAULT.walk(walker, tree);
-
+function parseAndRunListener(codeToParse: string) {
+    const parseResult = parseMyLangauge(codeToParse);
```

Comment:
Typo

Reply:
It is not clear to me what the error is I should correct on this line. Are you objecting to me wrapping the execution in function definition?